### ISSUE TITLE
Interplanetary Bodies - the 1.2 (or possibly 2.0) Plan.

### ISSUE THREAD

**Issue by severedsolo** [severedsolo/KerbinSpaceStation#27]
KSS is not going to scale well for interplanetary missions, so I propose a few changes (feedback welcome).

**All Missions:**
Tier 1 and 2 to be merged. Core mission has separate parameters for Power and Hab Modules.

**Interplanetary**
- New Mission: Dock a Service Module capable of landing on parent body (include LS Resupply in this mission)
- New Mission: Surface Survey (replaces ScienceExperimentModule) (Scientist only?)
- Evac Mission requires emergency landing on parent body.
- Medical Emergency Evac mission (futrtrubl)
- Require all crew members to return home each transfer window (ie proper crew rotation) and a new crew be sent to replace them.
- Require engineer on station with each Crew Rotation
- Fuel Resupply Missions?

**Tier 1 and 2 to be merged.**
_Problem:_
Launching 4 modules to attach to interplanetary stations, with one module generating at a time (therefore one module per transfer window), is going to get pretty tedious. For Duna alone it's going to be a minimum of 9-13 years to get a station built. Then you have to wait for another window to send your crew.

_Solution:_
I propose removing Tier 2 missions and instead expand the Core Mission with separate parameters for Power and Hab Modules. The player can then choose to either launch it all in one go, or make it modular. Also opens up the possibility for the person who requested Salyut/Mir type stations.

_Potential Issues:_
Old stations who haven't completed Tier 2 would be obsoleted, and could never advance. Can get around this by redefining vessels with a new name, but existing players would have to run the new Core Mission.

**New Missions**
_Problem:_
Interplanetary station missions should focus on parent bodies. It's unrealistic to expect the player to launch new modules/extra things needed for missions every transfer window. However, it's not unreasonable to expect player to crew rotate every transfer window.

_Solution:_
All interplanetary stations should need engineers at all times, in case the power module fails. The engineer can then repair it (or keep it going until a replacement arrives). (futrtrubl)

Evac missions provide an opportunity though: If the player has a Service Module (need to come up with a better name) capable of landing, and returning to the station, potentially the player could evacuate the crew to the surface, then back up when the emergency is over. Also the possibility of adding emergency medical evac missions (futrtrubl).

I also think it's unreasonable to expect the player to return science to Kerbin each time... so Survey missions could be used to land, collect science, return to the station, then transmit (EVA reports maybe?) 

This also opens up the possibility of Fuel Resupply Missions (the service module needs to be refuelled after each landing after all). Possibly combine these with the usual LS resupply missions (although I would expect the player to bring a load of LS with them).

_Potential Issues:_
1 Tier 3 mission at a time (as it is now) is a bit unrealistic. It would be better to allow players to (for example) do a Crew Rotation, Fuel Resupply, and LS Resupply mission on the same craft. Also, if I implement a change and it doesn't work well, I have to update each body (time consuming), so use Duna as the Beta test before implementing other bodies.

**Comment by futrtrubl** [comment]
I would have the station require an Engineer as crew and crew rotations to include an engineer. Makes sense if your station is so far away you can't get replacements quickly.
I think you can keep the replace mission, just say the engineer is doing the best he can but the module needs replacing at the next window. And then...
Combine Crew Rotation, Resuply, Replace and expansion missions together in the way the optional parts of the stock missions are put together so. 100% crew rotation + 50% chance of resuply + 50% chance of replace +50% chance of expansion.
I think the evac missions can be split in 2. One evac to surface due to unsafe conditions and then back. And one medical emergency evac to Kerbin.

**Comment by severedsolo** [comment]
Good ideas! Can't believe I didn't think of the engineer ones. I've updated the top post with your ideas. Not sure about the 50% resupply etc (not that I don't like the idea, I'm just not sure this is possible with CC)

**Comment by mcirish3** [comment]
Looks good to me for the most part.  One thing about interplanetary transfer windows is they are fairly broad.  The change in delta V over a couple of days during the optimal time frames is relatively small I see no reason a multi launch method for building the station on a single go could not easily be done. Though of course modular is always my favorite way to go.

**Comment by jrossignol** [comment]
You may want to look at using expressions for this - I think it could prevent you from having to write the same contract over and over again for different bodies.  I don't have much time to comment at the moment - let me know if you want some more details on this (I'll probably also come back later with some other thoughts).

**Comment by metl** [comment]
I had considered that but wasn't sure how to make it work with the DATA and expression node used to prevent multiple core launches to the same body.

**Comment by jrossignol** [comment]
One option is to duplicate the core launch contracts (ie. one per body).  Then for the other contracts, make them non-duplicated, and do something like this:

```
DATA
{
    type = Vessel
    targetVessel = [ DunaStation, EveStation, ... ].Where(v => v != null).Random()
}
```

Of course other things like limiting the number of contracts can get trickier...  but presumably the player isn't going to have more than 2-3 of these stations around the solar system....  presumably. :wink:

**Comment by severedsolo** [comment]
You know, that would work REALLY well for emergency missions (evac, repair/replace etc.) I could incorporate it into the Kerbin missions too. When I actually get round to implementing something (this weekend maybe...) 

I may have a play with that. With the core mission now covering everything, it would be much easier to have the main core missions running separately for each body (and would keep the "contract swamping" problem under control, even though you fixed that in 0.7.12).

I may come bothering you about this in a couple of days, if I can't get it working.

**Comment by severedsolo** [comment]
Fixed.

**Comment by severedsolo** [comment]
Implemented in 3.0
